src/sets/signerSet.ts

```diff
@@ -11,79 +12,75 @@ import { hashCompare } from '~/utils';
-  (2) vertexRemoves: remove set for vertices
-  (3) edgeAdds: add set for edges
-  (4) edgeremoves: remove set for edges
-  
-  The design diverges from a conventional 2P2P set in two ways:
-
-  (1) Vertices and edges are moved between sets, so a given vertex or edge can only exist in 
-      its respective adds or removes set 
-  (2) The edges set store the edge tuple (parentKey, childKey) as well as the hash of the SignerAdd
-      message that added that edge to the graph
-
-  In practice, there is a constraint that the edges add set is a tree and the edges 
-  remove set is a rooted directed graph. But that constraint is only a bi-product of the add
-  and remove methods, and it is not enforced by the data structure. This implementation 
-  favors keeping the edgeAdds and edgeRemoves data structure the same, which makes the sets
-  easier to compare. But this decision has two downsides:
-
-  (1) Tree and edge traversal is not as efficient as it could be. See the getEdgesByChild 
-      and getEdgesbyParent methods which filter keys in the edges maps by the child or parent 
-      members of the tuple.
-  (2) At points in the code where we need to get the parent of a vertex b in edgeAdds, we
-      loop through edges in edgeAdds where (*,b) even though practically we know there is only
-      one edge that fits that criteria. See the removeSubtree method for an example.
+  (1) custodyAdds: add set for custody addresses
+  (2) custodyRemoves: remove set for custody addresses
+  (3) signerAdds: add set for delegate signers
+  (4) signerRemoves: remove set for delegate signers
+
+  Each message is signed by a custody address that was added at a particular block number.
+  Conflicts are resolved in this order:
+  - Higher custody block number wins
+  - If custody block numbers are the same
+    - Removes win
+    - If two messages have the same type (i.e. both adds)
+      - Custody address with the higher lexicographical order wins
+      - If custody addresses have the same order (i.e. they are identical)
+        - Message with higher lexicographical hash wins
 */
-class SignerSet {
-  private _custodyAdds: Set<string>; // custody address
-  private _custodyRemoves: Set<string>; // custody address
-  private _vertexAdds: Set<string>; // pubKey
-  private _vertexRemoves: Set<string>; // pubKey
-  private _edgeAdds: Map<string, string>; // <parentKey, childKey>, hash
-  private _edgeRemoves: Map<string, string>; // <parentKey, childKey>, hash
-  private _messages: Map<string, SignerMessage>; // message hash => SignerAdd | SignerRemove
+
+export type SignerSetEvents = {
+  addCustody: (custodyAddress: string) => void;
+  removeCustody: (custodyAddress: string) => void;
+  addSigner: (signerKey: string) => void;
+  removeSigner: (signerKey: string) => void;
+};
+
+class SignerSet extends TypedEmitter<SignerSetEvents> {
+  private _custodyAdds: Map<string, CustodyAddEvent>;
+  private _custodyRemoves: Map<string, CustodyRemoveAll>;
```

Comment:
nit: it might be worth adding a comment here that this maps *every* removed address to a *CustodyRemoveAll* message and that there may be duplicate  remove all messages